kong/plugins/key-auth/handler.lua

```diff
@@ -90,13 +92,21 @@ local function do_authentication(conf)
     return nil, { status = 500, message = "Invalid plugin configuration" }
   end
 
+  if conf.validate_signature then
+    if type(conf.signature_names) ~= "table" then
+      kong.log.err("no conf.signature_names set, aborting plugin execution")
+      return nil, { status = 500, message = "Invalid plugin configuration" }
+    end
+  end
```

Comment:
I see you copied the one from above, but this should be part of the schema validation imo.